docs/standard/security/cross-platform-cryptography.md

```diff
@@ -0,0 +1,361 @@
+
+#### DSA native interop
+
+.NET exposes types to allow programs to interoperate with the OS libraries that the .NET cryptography code uses. The types involved don't translate between platforms and should only be directly used when necessary.
+
+| Type                                                         | Windows | Linux         | macOS         |
+|--------------------------------------------------------------|---------|---------------|---------------|
+| <xref:System.Security.Cryptography.DSACryptoServiceProvider> | ✔️     | ⚠️<sup>1</sup> | ⚠️<sup>1</sup> |
+| <xref:System.Security.Cryptography.DSACng>                   | ✔️     | ❌             | ❌            |
+| <xref:System.Security.Cryptography.DSAOpenSsl>               | ❌      | ✔️            | ⚠️<sup>2</sup> |
+
+<sup>1</sup> On macOS and Linux, <xref:System.Security.Cryptography.DSACryptoServiceProvider> can be used for compatibility with existing programs. In that case, any method that requires system interop, such as opening a named key, throws a <xref:System.PlatformNotSupportedException>.
+
+<sup>2</sup> On macOS, <xref:System.Security.Cryptography.DSAOpenSsl> works if OpenSSL is installed and an appropriate libcrypto dylib can be found via dynamic library loading. If an appropriate library can't be found, exceptions will be thrown.
+
+## X.509 Certificates
+
+The majority of support for X.509 certificates in .NET comes from OS libraries. To load a certificate into an <xref:System.Security.Cryptography.X509Certificates.X509Certificate2> or <xref:System.Security.Cryptography.X509Certificates.X509Certificate> instance in .NET, the certificate must be loaded by the underlying OS library.
+
+### Read a PKCS12/PFX
+
+| Scenario                                     | Windows | Linux | macOS |
+|----------------------------------------------|---------|-------|-------|
+| Empty                                        | ✔️     | ✔️    | ✔️   |
+| One certificate, no private key              | ✔️     | ✔️    | ✔️   |
+| One certificate, with private key            | ✔️     | ✔️    | ✔️   |
+| Multiple certificates, no private keys       | ✔️     | ✔️    | ✔️   |
+| Multiple certificates, one private key       | ✔️     | ✔️    | ✔️   |
+| Multiple certificates, multiple private keys | ✔️     | ❌    | ✔️   |
+
+### Write a PKCS12/PFX
+
+| Scenario                                     | Windows | Linux | macOS |
+|----------------------------------------------|---------|-------|-------|
+| Empty                                        | ✔️     | ✔️    | ❌   |
+| One certificate, no private key              | ✔️     | ✔️    | ❌   |
+| One certificate, with private key            | ✔️     | ✔️    | ✔️   |
+| Multiple certificates, no private keys       | ✔️     | ✔️    | ❌   |
+| Multiple certificates, one private key       | ✔️     | ✔️    | ✔️   |
+| Multiple certificates, multiple private keys | ✔️     | ❌    | ✔️   |
+| Ephemeral loading                            | ✔️     | ✔️    | ❌   |
```

Comment:
Ooh, something to update: All of the Xs in the read PKCS12 and write PKCS12 tables are checkmarks in .NET 5.

Maybe they should get warning glyphs with a footnote that it's available in .NET 5 previews, then turn into checkmarks after net5 RTM?